changelog.d/11547.bugfix

```diff
@@ -0,0 +1 @@
+Fix a bug introduced in Synapse 1.17.0 where a pusher created for an email with capital letters would fail to be created.
```

Comment:
#7021 added the canonical email stuff (I think) which was in Synapse 1.17.0, I'm not 100% sure what the behavior was before that though.